sphinx/writers/latex.py

```diff
@@ -841,10 +841,15 @@ def depart_rubric(self, node: Element) -> None:
     def visit_footnote(self, node: Element) -> None:
         self.in_footnote += 1
         label = cast(nodes.label, node[0])
+        if 'auto' not in node.attributes:
+            self.body.append('\\sphinxstepexplicit ')
```

Comment:
This will increment counter `sphinxexplicit`. In turn this will change `\thesphinxscope`. So never duplicate labels are created.

Reply:
How about using the pair of docname and node_id instead? All footnotes already have their original node_id. So we can make a unique id from the combination with docname (ex. `fn:{docname}:{node_id}`).

I understand this counter also helps to generate a unique id. But it makes LaTeX output a bit complicated to me (this feeling may come from my experience of python and docutils).

Reply:
The LaTeX ouput with ``\sphinxthescope`` was in preparation for in future easying up resetting numbering of footnotes per chaper, section or page. However I will review the rationale for it and follow your suggestion. Right now I feel there was some necessity on LaTeX side, but I need to remember why.

Reply:
@tk0miya The problem is with our ``visit_footnotemark()``. It uses minimal LaTeX markup ``\sphinxfootnotemark[27]`` for example. Then LaTeX must know from this ``27`` what was the label used for original footnote. The way it is done is via maintaining information in the form of two numeric counters: `sphinxscope` (but was introduced only fror future usage), and `sphinxexplicit`. At any given time in latex document `\thesphinxscope` expands to `<num1>.<num2>` where `num1` is "scope" (always 0 currently) and `num2` is the number last modified when an explicit ``footnote`` was encountered. So the three digits `num1` (scope), `num2` (count of explicit footnotes), `N` (from `\sphinxfootnotemark[N]`)  allow to build a label which will correspond to a unique anchor point. I agree it is involved. Originally I was using only `N` but I had overlooked the problem that `N` can be the same when an explictly numbered footnote is referred to multiple times. So I added the `sphinxexplicit` and `\sphinxstepexplicit`  at each explicit footnote to increment it. In the meantime I had also add `sphinxscope` but only as abstraction for future usage.

I can remove `scope` part and only ever refer `explicit` part.

Reply:
Originally, my idea was using the pair of docname and node_id via `\label` system. So footnotemark node will be rendered as `\sphinxfootnotemark[27]{fn:docname:idXX}`, for example (like `\ref`). As a result, we don't need to use counters on LaTeX output. But, as I commented above, both footnotetext and footnotemark nodes don't have their node_id (or referencing node_id).